source/serverless-testing/tests/README.rst

```diff
@@ -0,0 +1,115 @@
+tests that drivers MUST use to assert compatibility with Serverless.
+
+Test Configuration
+==================
+
+These tests MUST be run against a live Serverless instance on Atlas. A new
+Serverless instance MUST be created each time the test suite is run, and it MUST
+be deleted once the tests have completed, regardless of their outcome. The
+`serverless directory in the drivers-evergreen-tools repository`_ contains
+scripts for creating and deleting Serverless instances, and the ``config.yml``
+contains an example evergreen configuration that uses them to run the tests.
+
+.. _serverless directory in the drivers-evergreen-tools repository: https://github.com/mongodb-labs/drivers-evergreen-tools/tree/master/.evergreen/serverless
+
+All tests MUST be run with wire protocol compression and authentication
+enabled. Additionally, Serverless requires the usage of the versioned API, so
+the tests MUST be run with a server API version specified.
+
+Required Variables
+~~~~~~~~~~~~~~~~~~
+
+Managing the Serverless instances and connecting to them requires a few
+variables to be specified. The variables marked "private" are confidential and
+MUST be specified as private Evergreen variables or used only in private
+Evergreen projects. If using a public Evergreen project, xtrace MUST be disabled
+when using these variables to help prevent accidental leaks.
+
+- ``${SERVERLESS_DRIVERS_GROUP}``: Contains the ID of the Atlas group dedicated
+  to drivers testing of Serverless.
+
+- ``${SERVERLESS_API_PUBLIC_KEY}``: The public key required to use the Atlas API
+  for provisioning of Serverless instances.
+
+- ``${SERVERLESS_API_PRIVATE_KEY}``: (private) The private key required to use
+  the Atlas API for provisioning of Serverless instances.
+
+- ``${SERVERLESS_ATLAS_USER}``: (private) The SCRAM username used to
+  authenticate to any Serverless instance created in the drivers testing Atlas
+  group.
+
+- ``${SERVERLESS_ATLAS_PASSWORD}``: (private) The SCRAM password used to
+  authenticate to any Serverless instance created in the drivers testing Atlas
+  group.
+
+
+Existing Spec Tests
+===================
+
+Tests defined in the following specifications MUST be included in a driver's
+serverless testing suite, including prose tests:
+
+- CRUD, including the v1, v2, and unified tests
+
+In the future, this list will be expanded to include a greater portion of the
+tests once the Serverless proxy has more robust failCommand support.
+
+Note that the formats for the JSON/YAML tests of these specifications were
+updated to include a new ``runOnRequirement`` specifically for serverless
+testing, so make sure to update the affected test runners to support this
+requirement and then sync the tests.
+
+The Serverless proxy presents itself as a mongos, so any test meant to run
+against sharded clusters will be executed by the runners, with the exception of
+the tests affected by the previously mentioned ``runOnRequirement``.
+
+Prose Tests
```

Comment:
killCursors tests required by the scope. These are heavily modeled after similar tests for Atlas Data Lake.